src/lib.rs

```diff
@@ -2458,6 +2458,54 @@ where
     go_extra!(O);
 }
 
+impl<'a, I, O, E, T> ParserSealed<'a, I, O, E> for Box<T>
+where
+    I: Input<'a>,
+    E: ParserExtra<'a, I>,
+    T: Parser<'a, I, O, E>,
+{
+    fn go<M: Mode>(&self, inp: &mut InputRef<'a, '_, I, E>) -> PResult<M, O>
+    where
+        Self: Sized,
+    {
+        T::go::<M>(self, inp)
+    }
+
+    go_extra!(O);
+}
+
+impl<'a, I, O, E, T> ParserSealed<'a, I, O, E> for ::std::rc::Rc<T>
```

Comment:
Might want to use `::core::rc::Rc` here so we're still `no_std`. Sample with the others.